clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -92,6 +93,15 @@ public MetadataRequest build(short version) {
             if (!data.allowAutoTopicCreation() && version < 4)
                 throw new UnsupportedVersionException("MetadataRequest versions older than 4 don't support the " +
                         "allowAutoTopicCreation field");
+            if (version >= 10) {
```

Comment:
Is this version check needed at all?

Reply:
We will need to check the version when this is fixed, but I can remove the version check for now.

Reply:
+1